### PR TITLE
Support isEmptyValue flag for metrics query.

### LINKED ISSUE #10501: [Feature] Support isEmptyValue flag for metrics query
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

In the current case, we always respond to zero if timestamp slots are absent.

I opened https://github.com/apache/skywalking-query-protocol/pull/110 and https://github.com/apache/skywalking-query-protocol/pull/112 for a protocol update(9.5.0 only), and we could distinguish these two cases. 

### Use case

`no value`(no metric at this time bucket) V.S. `actual zero value`(such as success rate == 0% or the value is less than 1 due to precision)

@wankai123 According to this, let's evaluate whether the default value still makes sense, or we should remove that permanently. 

FYI @Fine0830 After this lands on the backend, UI should do
1. Build a flat line cross the absent(isEmptyValue = true) data point. Rather than dropping the line to zero as a peak. 
2. Skip absent(isEmptyValue = true) data point when the table runs preview avg.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wankai123**
And Fix `NPE` when query the not exist series indexes in ElasticSearch storage.

- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [X] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #10501.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: test/e2e-v2/cases/php/expected/metrics-has-value.yml [hunk 1]
```diff
@@ -15,5 +15,11 @@
 
 {{- contains . }}
 - key: {{ notEmpty .key }}
-  value: {{ ge .value 2 }}
+  value:
+    value: 0
+    isemptyvalue: true
```
Comment (kezhenxu94) [inline test/e2e-v2/cases/php/expected/metrics-has-value.yml:20]: Why do we need to assert that there is empty value?

Comment (wu-sheng) [reply test/e2e-v2/cases/php/expected/metrics-has-value.yml:20]: I think the point is, from a test perspective, the initial query should cover the time range from no telenmetry data to having data.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/MetricsQueryService.java [hunk 2]
```diff
@@ -50,7 +51,7 @@ private IMetricsQueryDAO getMetricQueryDAO() {
     /**
      * Read metrics single value in the duration of required metrics
      */
-    public long readMetricsValue(MetricsCondition condition, Duration duration) throws IOException {
+    public NullableValue readMetricsValue(MetricsCondition condition, Duration duration) throws IOException {
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/MetricsQueryService.java:54]: I think we don't change this API. Somehow we don't tested this?

We just added a new API --> **readNullableMetricsValue**

**Review by wu-sheng — APPROVED** [review]
LGTM

**Pull request merged**